common/download.go

```diff
@@ -278,20 +278,27 @@ func (d *HTTPDownloader) Download(dst *os.File, src *url.URL) error {
 	}
 
 	resp, err := httpClient.Do(req)
-	if err == nil && (resp.StatusCode >= 200 && resp.StatusCode < 300) {
-		// If the HEAD request succeeded, then attempt to set the range
-		// query if we can.
-		if resp.Header.Get("Accept-Ranges") == "bytes" {
-			if fi, err := dst.Stat(); err == nil {
-				if _, err = dst.Seek(0, os.SEEK_END); err == nil {
-					req.Header.Set("Range", fmt.Sprintf("bytes=%d-", fi.Size()))
-
-					d.current = uint64(fi.Size())
+	if err == nil {
+
+		if resp.StatusCode >= 200 && resp.StatusCode < 300 {
+			// If the HEAD request succeeded, then attempt to set the range
+			// query if we can.
+			if resp.Header.Get("Accept-Ranges") == "bytes" {
+				if fi, err := dst.Stat(); err == nil {
+					if _, err = dst.Seek(0, os.SEEK_END); err == nil {
+						req.Header.Set("Range", fmt.Sprintf("bytes=%d-", fi.Size()))
+
+						d.current = uint64(fi.Size())
+					}
 				}
 			}
 		}
-	} else if err != nil || (resp.StatusCode >= 400 && resp.StatusCode < 600) {
-		return fmt.Errorf("%s", resp.Status)
+
+		if resp.StatusCode >= 400 && resp.StatusCode < 600 {
+			return fmt.Errorf("Received HTTP error: %s", resp.Status)
+		}
+	} else {
+		return err
```

Comment:
I don't think we want to return the error here. This is just a HEAD request and we don't really care if it fails. That's why we didn't check for the error, originally.

I think something like 

```go
if err == nil 
  if resp.StatusCode >= 200 && resp.StatusCode < 300 {
    ...
  } else {
    log.Printf("[DEBUG] (download) Unexpected HTTP response during HEAD request: %s", resp.Status)
} else {
  log.Printf("[DEBUG] (download) Error making HTTP HEAD request: %s", err.Error())
}
```

Reply:
That makes total sense, thanks for pointing that out. I've made a new commit (flipped the logic a little bit and addressed the GET request code).